### PR TITLE
Avoid importing `k3sHelper` from UI

### PR DESCRIPTION
**Pull request by mook-as**
Because `k3sHelper.ts` pulls in `@pkg/utils/paths` (which is main-process only and spawns `rdctl`), we must not use it from the renderer process as that will cause an error when the relevant window opens.  Instead, move the Kubernetes version helpers into a new file that does not import it.

While we're at it, avoid sending `semver.SemVer` objects across processes because that does not work anyway.  See #7208 (though we still need to do an audit to make sure there are no other non-cloneable objects being passed).

### CODE REVIEW

**Review by jandubois — DISMISSED** [review]
I'm not sure if this PR is ok; please double-check if the change from `mgr` to `k8smanager` was intention or a copy&pasto!

If that was intentional, then I'm fine with merging, although ideally I would like to remove `k8s.VersionInfo` completely.

File: pkg/rancher-desktop/backend/k8s.ts [hunk 1]
```diff
@@ -14,15 +15,7 @@ export type {
...
-/**
- * VersionEntry describes a version of K3s.
- */
-export interface VersionEntry {
-  /** The version being described. This includes any build-specific data. */
-  version: semver.SemVer;
-  /** A string describing the channels that include this version, if any. */
-  channels?: string[];
-}
+export type { SemanticVersionEntry as VersionEntry } from '@pkg/utils/kubeVersions';
```
[the remark this answers is not in the document]
Comment (mook-as) [reply pkg/rancher-desktop/backend/k8s.ts]: Will do.

File: pkg/rancher-desktop/utils/kubeVersions.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import semver from 'semver';
+
+export interface VersionEntry {
+  /**
+   * The version being described. This includes any build-specific data.
+   * This must be a valid semver-parsable string, without any pre-release
+   * versions or build metadata.
+   */
+  version: string;
+  /** A string describing the channels that include this version, if any. */
```
[the remark this answers is not in the document]
Comment (mook-as) [reply pkg/rancher-desktop/utils/kubeVersions.ts]: Good catch, changed. (It was copied from `k8s.ts` where it was already wrong.)

File: background.ts [hunk 3]
```diff
@@ -827,7 +827,10 @@ ipcMainProxy.on('k8s-restart', async() => {
 });
 
 ipcMainProxy.on('k8s-versions', async() => {
-  window.send('k8s-versions', await k8smanager.kubeBackend.availableVersions, await k8smanager.kubeBackend.cachedVersionsOnly());
+  const versions = await k8smanager.kubeBackend.availableVersions;
+  const cachedOnly = await k8smanager.kubeBackend.cachedVersionsOnly();
+
+  window.send('k8s-versions', versions.map(e => ({ version: e.version.version, channels: e.channels })), cachedOnly);
```
Comment (jandubois) [inline background.ts:833]: Should `SemanticVersionEntry` have a `versionEntry` method that returns the data as a `VersionEntry` object, and then you could write

```suggestion
  window.send('k8s-versions', versions.map(e => e.versionEntry), cachedOnly);
```

File: pkg/rancher-desktop/backend/k3sHelper.ts [hunk 4]
```diff
@@ -112,12 +112,12 @@ export class ChannelMapping {
 }
 
 /**
- * VersionEntry implements K8s.VersionEntry.
+ * SemanticVersionEntry implements K8s.SemanticVersionEntry.
```
Comment (jandubois) [inline pkg/rancher-desktop/backend/k3sHelper.ts:115]: Why is `k8s.SemanticVersionEntry` needed? Why can't we use the `kubeVersion.SemanticVersionEntry` everywhere?

File: pkg/rancher-desktop/backend/k8s.ts [hunk 5]
```diff
@@ -14,15 +15,7 @@ export type {
...
-/**
- * VersionEntry describes a version of K3s.
- */
-export interface VersionEntry {
-  /** The version being described. This includes any build-specific data. */
-  version: semver.SemVer;
-  /** A string describing the channels that include this version, if any. */
-  channels?: string[];
-}
+export type { SemanticVersionEntry } from '@pkg/utils/kubeVersions';
```
Comment (jandubois) [inline pkg/rancher-desktop/backend/k8s.ts:18]: Why do we need to export this type from `k8s`. Why can't we import from `kubeVersion` everywhere?

File: pkg/rancher-desktop/utils/kubeVersions.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+/**
+ * Get the highest patch release of the lowest available versions
+ * @param versions The list of K8s.VersionEntry objects.
+ * @returns The highest patch version.
+ */
+export function minimumUpgradeVersion(versions: SemanticVersionEntry[]): SemanticVersionEntry | undefined {
+  // See comment on highestStableVersion about versions[].version potentially not being a semver.SemVer object.
+  const lowestFirst = versions.slice().sort((a, b) => semver.compare(a.version, b.version));
```
Comment (jandubois) [inline pkg/rancher-desktop/utils/kubeVersions.ts:50]: The comment is now wrong. And we could simplify the comparison; not sure it makes any difference, but shorter is better...

```suggestion
  const lowestFirst = versions.slice().sort((a, b) => a.compare(b));
```

**Review by jandubois — DISMISSED** [review]
Thanks, LGTM

Still some stylistic feedback though.

File: background.ts (hunk 3 above)
Comment (mook-as) [reply background.ts:833]: Currently `SemanticVersionEntry` is an interface, so it doesn't have any methods. I can look into switching it to a concrete class, though.

**Review by jandubois — APPROVED** [review]
Thanks, LGTM

**Pull request merged**